libdevcore/CommonData.h

```diff
@@ -42,22 +42,44 @@ enum class WhenError
 	Throw = 1,
 };
 
-enum class HexPrefix
+namespace
 {
-	DontAdd = 0,
-	Add = 1,
-};
-/// Convert a series of bytes to the corresponding string of hex duplets.
-/// @param _w specifies the width of the first of the elements. Defaults to two - enough to represent a byte.
+
+const char* hexdigits = "0123456789abcdef";
+
+template <class T>
+inline void writeHex(T const& _data, std::string& o_hex, size_t _off)
+{
+	for (byte i: _data)
+	{
+		o_hex[_off++] = hexdigits[i>>4];
+		o_hex[_off++] = hexdigits[i&0x0f];
+	}
+}
+
+}
+
+/// Convert a series of bytes to the corresponding hex string.
 /// @example toHex("A\x69") == "4169"
 template <class T>
-std::string toHex(T const& _data, int _w = 2, HexPrefix _prefix = HexPrefix::DontAdd)
```

Comment:
I like removing `_w` & having separate function instead of `_prefix` but I remember @chfast resising the similar change in one of the recent PRs